dotnet/src/IntegrationTests/Connectors/AzureAIInference/AzureAIInferenceChatCompletion_FunctionCallingTests.cs

```diff
@@ -0,0 +1,362 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using SemanticKernel.IntegrationTests.TestSettings;
+using Xunit;
+
+using ChatMessageContent = Microsoft.SemanticKernel.ChatMessageContent;
+
+namespace SemanticKernel.IntegrationTests.Connectors.AzureAIInference;
+
+public sealed class AzureAIInferenceChatCompletionFunctionCallingTests : BaseIntegrationTest
+{
+    // Complex parameters currently don't work (tested against llama3.2 model)
+    [Fact(Skip = "For manual verification only")]
+    public async Task CanAutoInvokeKernelFunctionsWithComplexTypeParametersAsync()
+    {
+        // Arrange
+        var kernel = this.CreateAndInitializeKernel();
+        kernel.ImportPluginFromFunctions("HelperFunctions",
+        [
+            kernel.CreateFunctionFromMethod((WeatherParameters parameters) =>
+            {
+                if (parameters.City.Name == "Dublin" && (parameters.City.Country == "Ireland" || parameters.City.Country == "IE"))
+                {
+                    return Task.FromResult(42.8); // 42.8 Fahrenheit.
+                }
+
+                throw new NotSupportedException($"Weather in {parameters.City.Name} ({parameters.City.Country}) is not supported.");
+            }, "Get_Current_Temperature", "Get current temperature."),
+        ]);
+
+        PromptExecutionSettings settings = new() { FunctionChoiceBehavior = FunctionChoiceBehavior.Auto() };
+
+        // Act
+        var result = await kernel.InvokePromptAsync("What is the current temperature in Dublin, Ireland, in Fahrenheit?", new(settings));
+
+        // Assert
+        Assert.NotNull(result);
+        Assert.Contains("42.8", result.GetValue<string>(), StringComparison.InvariantCulture); // The WeatherPlugin always returns 42.8 for Dublin, Ireland.
+    }
+
+    [Fact]//(Skip = "For manual verification only")]
```

Comment:
Double checking if these tests should be enabled by default. If yes, maybe it's worth to remove the comment. Same in other places.

Reply:
Good catch, added Skip, as currently I wasn't able to deploy any model in Azure AI Studio capable of function calling, including llama3.2 with a weird error message:

```http
HTTP/1.1 400 Bad Request
Content-Length: 246
Content-Type: application/json
x-ms-rai-invoked: false
x-request-id: b0bf8687-9587-4cc6-8cd8-66d2bf86e22d
ms-azureml-model-error-reason: model_error
ms-azureml-model-error-statuscode: 400
x-ms-client-request-id: 4796effe-4005-40fd-aaf1-d5ddd19159fd
Request-Context: appId=cid-v1:38659f32-9fc9-4714-92e6-1412a97fbbfc
azureml-model-session: dep-90b-vision-240926224530
azureml-model-group: llama-3-2-90b-scaleset
Date: Mon, 11 Nov 2024 15:39:51 GMT
Connection: close

{  
"error": {
    "code": "Bad Request",
    "message": "{\"object\":\"error\",\"message\":\"\\\"auto\\\" tool choice requires --enable-auto-tool-choice and --tool-call-parser to be set\",\"type\":\"BadRequestError\",\"param\":null,\"code\":400}",
    "status": 400
  }
}
```